src/status_im2/subs/contact_test.cljs

```diff
@@ -0,0 +1,185 @@
+              :contact-request-state 1})}
+    {:title "R"
+     :data  ({:active?               true
+              :last-updated          1672582629695
+              :mutual?               true
+              :blocked?              false
+              :contactRequestClock   0
+              :images                {}
+              :added                 true
+              :name                  "slim.shady"
+              :added?                true
+              :Removed               false
+              :trustStatus           0
+              :alias                 "Real Slim Shady"
+              :bio                   ""
+              :IsSyncing             false
+              :display-name          ""
+              :ens-verified          true
+              :socialLinks           nil
+              :blocked               false
+              :allow-new-users?      true
+              :verificationStatus    0
+              :lastUpdatedLocally    1672582563204
+              :public-key            "0xtest"
+              :names                 {:nickname         nil
+                                      :display-name     ""
+                                      :three-words-name "Real Slim Shady"
+                                      :ens-name         "@slim.shady"}
+              :has-added-us          true
+              :contact-request-state 1})}))
+
+(h/deftest-sub :contacts/sorted-and-grouped-by-first-letter
+  [sub-name]
+  (testing "Returning empty sequence when no contacts"
+    (swap! rf-db/app-db merge (assoc contacts-sample-data :contacts/contacts {}))
+    (is (empty? (rf/sub [sub-name]))))
+  (testing "Returning empty sequence when no mutual contacts"
+    (let [remove-contact-as-mutual #(-> %
+                                        (assoc-in ["0xtest" :mutual?] false)
+                                        (assoc-in ["0xtest2" :mutual?] false)
+                                        (assoc-in ["0xtest3" :mutual?] false))]
+      ;; TODO(@ibrkhalil,2023-01-01): Replace with update-vals when we update the ClojureScript API
+      (swap! rf-db/app-db merge
+        (update contacts-sample-data :contacts/contacts remove-contact-as-mutual))
+      (is (empty? (rf/sub [sub-name])))))
+  (testing "Returning sorted contacts"
+    (swap! rf-db/app-db merge contacts-sample-data)
+    (let [contact-list-without-identicons (map (fn [contact-group]
+                                                 (update contact-group
+                                                         :data
+                                                         #(map (fn [contact]
+                                                                 (dissoc contact :identicon))
+                                                               %)))
+                                               (rf/sub [sub-name]))]
+      ;; TODO(@ibrkhalil,2023-01-01): Find out why identicons get auto generated timestamps on their
```

Comment:
I don't quite grasp this TODO @ibrkhalil. Removing the identicons is necessary because we're writing a deterministic test, but the test data has random bits.

You did the right thing here, so to me the TODO should be removed because we can't write an assertion on top of random data, unless we start to use more advanced (and custom) matchers.